internal/js-dependency-scanner/src/main/java/org/wso2/security/tools/scanner/dependency/js/preprocessor/ReactFileDownloader.java

```diff
@@ -0,0 +1,196 @@
+import org.wso2.security.tools.scanner.dependency.js.utils.HttpDownloadUtility;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.sql.Timestamp;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * ResourceDownloader to download package.json files. It reads the pom file from product repository
+ * then get the component version where those package.json files need to be downloaded. After that download
+ * package.json files from particular version release tag of relevant component. These are performed using
+ * github api.
+ */
+public class ReactFileDownloader {
+    private static final Charset UTF_8 = StandardCharsets.UTF_8;
+
+    private static final Logger log = Logger.getLogger(ReactFileDownloader.class);
+
+    /**
+     * Download Package.json files.
+     *
+     * @param rootDirectory Root directory where package.json files to be downloaded.
+     * @throws DownloaderException Error occurred during Downloading react files.
+     */
+    public static void downloadReactFiles(String version, HashMap<String, String> repoVersionMapper, File rootDirectory,
+                                          String productRepo)
+            throws DownloaderException, ApiInvokerException {
+        for (Map.Entry<String, String> entry : repoVersionMapper.entrySet()) {
+            String componentVersion = getComponentVersionFromPom(version, productRepo, entry.getValue());
+            getPackagejsonFiles(componentVersion, entry.getKey(), rootDirectory);
+        }
+    }
+
+    /**
+     * Install npm modules which are referenced in package.json files.
+     *
+     * @param path Directory where the package.json file is placed.
+     * @throws DownloaderException exception occurred while downloading NPM Modules.
+     */
+    private static void installNPMModules(String path) throws DownloaderException {
+        Process process = null;
+        try {
+            ProcessBuilder processBuilder = new ProcessBuilder(JSScannerConstants.NPM,
+                    JSScannerConstants.INSTALL);
+            processBuilder.directory(new File(path));
+            process = processBuilder.start();
+            process.waitFor();
+        } catch (IOException | InterruptedException e) {
+            throw new DownloaderException("Error occurred while downloading NPM modules due to : " + e.getMessage());
+        } finally {
+            assert process != null;
+            process.destroyForcibly();
+        }
+    }
+
+    /**
+     * Create Directory
+     *
+     * @param dir Directory to be created
+     */
+    private static void createDirectory(File dir) {
+
+        if (!dir.exists()) {
+            boolean isDirCreated = dir.mkdir();
+            if (!isDirCreated) {
+                log.error((dir.getAbsolutePath() + " is not created"));
```

Comment:
Should the caller of this method notified regarding the failure to create the directory?